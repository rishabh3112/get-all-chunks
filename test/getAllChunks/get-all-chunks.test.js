const { getAllChunks } = require("../../lib/utils/getAllChunks.js");
const { resolve } = require("path");

const testPath = resolve(__dirname, "../src/code.js");

describe("basic", () => {
  it("should output correct dependency tree", async () => {
    const depTree = await getAllChunks(testPath);
    expect(depTree).toMatchSnapshot({
      children: expect.any(Array),
      chunks: expect.any(Set),
      path: expect.any(String),
    });
  });

  it("should output correct set of chunks", async () => {
    const depTree = await getAllChunks(testPath);
    expect(depTree.path).toBe(testPath);
    expect(depTree.chunks.size).toBe(2);
    expect(depTree.chunks).toMatchInlineSnapshot(
      `
Set {
  "${resolve(__dirname, "../src/folderc/c.js")}",
  "${resolve(__dirname, "../src/folderb/b.js")}",
}
`
    );
  });
});
