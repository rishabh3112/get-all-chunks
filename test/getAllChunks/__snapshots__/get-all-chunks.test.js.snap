// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic should output correct dependency tree 1`] = `
Object {
  "children": Any<Array>,
  "chunks": Any<Set>,
  "path": Any<String>,
}
`;
